homeassistant/config_entries.py

```diff
@@ -3485,7 +3485,22 @@ async def async_finish_flow(
         entry = self.hass.config_entries.async_get_known_entry(flow.handler)
 
         if result["data"] is not None:
-            self.hass.config_entries.async_update_entry(entry, options=result["data"])
+            automatic_reload = False
+            if isinstance(flow, OptionsFlowWithReload):
```

Comment:
If we have a subclass for handling a specific behavior, I think it would make sense for that behavior to be implemented in the subclass instead of making the base class know about the subclass and handle it.

I think this could be done in one of two ways:

- define `automatic_reload = False` in `OptionsFlow`, and make `OptionsFlowWithReload` overrides it to `True`.
- move the reload implementation to `OptionsFlowWithReload`, and override `async_finish_flow` so that all the reload logic stays on the subclass (it may require moving part of the code that is on `async_finish_flow` to a new private method).

I prefer the 2nd option, as it prevents the subclass behavior from creeping to the base class.

Reply:
The logic for handling the flow is in the flow manager, there is no subclassing of it.

Reply:
:man_facepalming: sorry, I completely missed that somehow. I should've checked the surrounding context instead of looking only at the diff here.